### PR TITLE
Add video upload support to the micropub endpoint

### PR DESCRIPTION
**Pull request by cweiske**
Requires a video plugin to be installed, e.g.
https://github.com/cweiske/withknownVideo

Related: #1374

### CODE REVIEW

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php [hunk 1]
```diff
@@ -111,14 +111,29 @@ function post()
...
                         if (!$success) {
                             \Idno\Core\Idno::site()->triggerEvent('indiepub/post/failure', ['page' => $this]);
                             $this->setResponse(500);
                             echo "Failed uploading photo from $photo_url";
                             exit;
                         }
-                    } else if (!empty($name)) {
+                    }
+
+                    if (!empty($_FILES['video'])) {
```
Comment (kylewm) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: `else if`? I'm assuming a post cannot have both a photo and a video

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php [hunk 2]
```diff
@@ -111,14 +111,29 @@ function post()
...
+                        $success   = $this->uploadFromUrl('video', $video_url);
+                        if (!$success) {
+                            \Idno\Core\Idno::site()->triggerEvent('indiepub/post/failure', ['page' => $this]);
+                            $this->setResponse(500);
+                            echo "Failed uploading video from $video_url";
+                            exit;
+                        }
+                    }
+
+                    if ($type == 'note' && !empty($name)) {
```
Comment (kylewm) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: If you use `else if` here too, you can remove the `$type == note` check

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php (hunk 1 above)
Comment (cweiske) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: Some micropub clients (ownyougram and shpub) upload a video and the preview image at the same time, so having both is actually used. Known throws the photo away right now, but that can be changed in the future.

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php (hunk 2 above)
Comment (cweiske) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: no, I can't because of above.

**Comment by benwerd** [comment]
This, again, looks like it could be really cool (although it would be good to have a POC video plugin that supports both images and video). I'm in favor of merging once conflicts have been resolved.

High five @cweiske for contributing to core!

**Comment by cweiske** [comment]
So what can I do to get this patch merged?

**Comment by physcocode** [comment]
@kylewm @cweiske I think this is ready to merge I see no conflict here . Ownyourgram actually uploads both image and preview at same time !

**Comment by kylewm** [comment]
Sorry for the delay; I didn't see that you'd fixed the merge conflicts :+1:

**Pull request merged**